internal/internal_worker_test.go

```diff
@@ -227,6 +227,14 @@ func (s *internalWorkerTestSuite) TestCreateWorker_WithAutoScaler() {
 	worker.Stop()
 }
 
+func (s *internalWorkerTestSuite) TestCreateWorker_WithStrictNonDeterminism() {
+	worker := createWorkerWithStrictNonDeterminismOption(s.T(), s.service)
+	err := worker.Start()
+	require.NoError(s.T(), err)
+	time.Sleep(time.Millisecond * 200)
```

Comment:
what's the sleep doing here?

Reply:
All the tests in this file have it. Removing sleep doesn't change the result when a single test case is run but when the whole suit is run without sleeps it gets `signal: interrupt`. Not going to deep dive here as it is totally irrelevant to this change. I am not fan of using test suite and assert libraries in Go. Feel free to take a look if you have time, we can discuss later.

Reply:
ah, yea.  following others with these suites is (unfortunately) reasonable.
I'd love to rip them all out, but it is a big task :|